hca/dss/util/__init__.py

```diff
@@ -51,3 +56,10 @@ def hardlink(source, link_name):
         # Check device first because different drives can have the same inode number
         if source_stat.st_dev != dest_stat.st_dev or source_stat.st_ino != dest_stat.st_ino:
             raise
+    except OSError as e:
+        if e.errno == errno.EMLINK:
+            # FIXME: Copying is not space efficient; see https://github.com/HumanCellAtlas/dcp-cli/issues/453
+            log.warning('Too many hard links to a file, reverting to copying')
```

Comment:
I'd log the file names unless that's obvious from a previous log statement. But given that this is run in multiple threads I doubt the user can easily correlate log lines from a single thread.